packages/nextjs/src/server/protect.ts

```diff
@@ -84,14 +123,28 @@ export function createProtect(opts: {
     };
 
     const handleUnauthorized = () => {
+      // For machine tokens, return a 401 response
+      if (authObject.tokenType !== TokenType.SessionToken) {
+        return unauthorized();
+      }
+
       if (unauthorizedUrl) {
         return redirect(unauthorizedUrl);
       }
       return notFound();
     };
 
-    if (authObject.tokenType !== 'session_token') {
-      throw new Error('TODO: Handle machine auth object');
+    if (!isTokenTypeAccepted(authObject.tokenType, requestedToken)) {
+      return handleUnauthorized();
+    }
+
+    if (authObject.tokenType !== TokenType.SessionToken) {
+      // For machine tokens, we only check if they're authenticated
+      // They don't have session status or organization permissions
+      if (!authObject.id) {
+        return handleUnauthorized();
+      }
+      return authObject;
     }
```

Comment:
Currently, `auth.protect()` either redirects to sign in page or renders a 404 Nextjs page.

Now, we're actually sending a 401 response only when:
- mismatched tokens
- failed machine verification (for oauth_tokens, it'll include a `www-authenticate` header)